doc/source/serve/production-guide/kubernetes.md

```diff
@@ -0,0 +1,242 @@
+See the [Kubernetes documentation](https://kubernetes.io/docs/setup/) or the [KubeRay quickstart guide](kuberay-quickstart) if you need help getting started.
+
+The first step is to install the `KubeRay` operator into your Kubernetes cluster.
+This creates a pod that runs the `KubeRay` controller. The `KubeRay` controller manages resources based on the `RayService` CRs you create.
+
+Install the operator using `kubectl apply` and check that the controller pod is running:
+```console
+$ kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/default?ref=v0.3.0&timeout=90s"
+$ kubectl get deployments -n ray-system
+NAME                READY   UP-TO-DATE   AVAILABLE   AGE
+kuberay-apiserver   1/1     1            1           13s
+kuberay-operator    1/1     1            1           13s
+
+$ kubectl get pods -n ray-system
+NAME                                 READY   STATUS    RESTARTS   AGE
+kuberay-apiserver-799bc6dd95-787w7   1/1     Running   0          42s
+kuberay-operator-68c75b5d5f-m8xd7    1/1     Running   0          42s
+```
+
+For more details, see the [KubeRay quickstart guide](kuberay-quickstart).
+
+## Deploying a Serve application
+
+Once the KubeRay controller is running, you can manage your Ray Serve application by creating and updating a `RayService` custom resource (CR).
+`RayService` custom resources consist of the following:
+- a `KubeRay` `RayCluster` config defining the cluster that the Serve application runs on.
+- a Ray Serve [config](serve-in-production-config-file) defining the Serve application to run on the cluster.
+
+When the `RayService` is created, the `KubeRay` controller first creates a Ray cluster using the provided configuration.
+Then, once the cluster is running, it deploys the Serve application to the cluster using the [REST API](serve-in-production-deploying).
+The controller also creates a Kubernetes Service that can be used to route traffic to the Serve application.
+
+Let's see this in action by deploying the [`FruitStand` example](serve-in-production-example).
+The Serve config for the example is embedded into [this example `RayService` CR](https://github.com/ray-project/kuberay/blob/release-0.3/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml).
+To follow along, save this CR locally in a file named `ray_v1alpha1_rayservice.yaml`:
+
+```console
+$ curl -o ray_v1alpha1_rayservice.yaml https://raw.githubusercontent.com/ray-project/kuberay/release-0.3/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml
+```
+
+To deploy the example, we simply `kubectl apply` the CR.
+This creates the underlying Ray cluster, consisting of a head and worker node pod (see [Ray Clusters Key Concepts](../../cluster/key-concepts.rst) for more details on Ray clusters), as well as the service that can be used to query our application:
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
```

Comment:
Could use to include a warning about resource requests/limits, along the lines of the one right above the bookmark here
https://docs.ray.io/en/master/cluster/kubernetes/getting-started.html#running-applications-on-a-ray-cluster
and the one in the config file here:
https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.complete.yaml#L1-L4

We don't want people accidentally burning themselves with a small head pod.

Reply:
done, added a link to kuberay config docs